src/ui/components/common/content_actions/report_button.rs

```diff
@@ -0,0 +1,45 @@
+use crate::{
+  ui::components::common::icon::{Icon, IconType},
+  utils::types::{ContentActionType, ReportModalData, ReportModalNode},
+};
+use leptos::*;
+
+fn report_content(id: i32, content_type: ContentActionType, creator_actor_id: String) {
+  let set_report_modal_data = expect_context::<WriteSignal<ReportModalData>>();
+  let report_modal = expect_context::<ReportModalNode>().0;
+
+  set_report_modal_data.set(ReportModalData {
+    id,
+    content_type,
+    creator_actor_id,
+  });
+  let _ = report_modal
+    .get_untracked()
+    .expect("Report dialog should exist")
+    .show_modal();
+}
+
+#[component]
+pub fn ReportButton(
+  id: i32,
```

Comment:
Same for all these IDs, make sure they're typed with `PostId`, `CommentId`, etc.

Reply:
[See my other relevent comment.](https://github.com/LemmyNet/lemmy-ui-leptos/pull/107#discussion_r1694362491)